internal/assets/manifests/istio-discovery/values.yaml.tpl

```diff
@@ -107,11 +107,34 @@ pilot:
 {{ $x | indent 2 }}
 {{- end }}
 
-{{- if .GetSpec.GetHttpProxyEnvs }}
 sidecarInjectorWebhook:
   # Supported only in Cisco provided istio-proxy images
 {{ toYamlIf (dict "value" .GetSpec.GetHttpProxyEnvs "key" "httpProxyEnvs") | indent 2 }}
+  templates:
+{{- if .GetSpec.GetSidecarInjector.GetTemplates.GetSidecar }}
+{{ valueIf (dict "key" "sidecarOverrides" "value" .GetSpec.GetSidecarInjector.GetTemplates.GetSidecar) | indent 4 }}
 {{- end }}
+{{- if not .GetSpec.GetSidecarInjector.GetTemplates.GetSidecar }}
+    sidecarOverrides: ""
+{{- end }}
+{{- if .GetSpec.GetSidecarInjector.GetTemplates.GetGateway }}
+{{ valueIf (dict "key" "gatewayOverrides" "value" .GetSpec.GetSidecarInjector.GetTemplates.GetGateway) | indent 4 }}
+{{- end }}
+{{- if not .GetSpec.GetSidecarInjector.GetTemplates.GetGateway }}
+    gatewayOverrides: ""
+{{- end }}
+{{- if .GetSpec.GetSidecarInjector.GetTemplates.GetCustomTemplates }}
+{{ range $customTemplate := .GetSpec.GetSidecarInjector.GetTemplates.GetCustomTemplates }}
+{{ valueIf (dict "key" $customTemplate.Name "value" $customTemplate.Template) | indent 4 }}
+{{ end }}
+{{- end }}
+{{- if .GetSpec.GetSidecarInjector.GetTemplates.GetSidecar }}
```

Comment:
this same if is used in line 114. maybe setting the `defaultTemplates` can be done as well?

Reply:
merged